poh/src/poh_recorder.rs

```diff
@@ -579,6 +582,8 @@ impl PohRecorder {
     }
 
     pub fn set_bank(&mut self, bank: &Arc<Bank>, track_transaction_indexes: bool) {
+        assert!(self.working_bank.is_none());
```

Comment:
Sanity check here on the logic of `PohRecorder`.

This potentially adds a new panic, but is a bit more direct than if we get the panic in notifier for having 2 in progress banks.